### PR TITLE
[MSEARCH-767].  Implement Reindexing of Campuses

### PR DESCRIPTION
**Pull request by BKadirkhodjaev**
### Purpose
Implement Reindexing of Campuses [[MSEARCH-767]](<https://folio-org.atlassian.net/browse/MSEARCH-767>)

### Approach
- Add mod-inventory-storage campus client
- Update campuses model to support campuses reindex
- Update reindex service to support campuses reindex
- Implement campuses reindexing by repeatedly querying campuses from mod-inventory-storage and indexing them
- Test manually using mod-search, mod-inventory-storage (master branch), Postman, Elasticsearch, Kibana (please follow the Jira ticket to see the test output).

### Changes Checklist
- [ ] **API Changes**: Document any API paths, methods, request or response bodies changed, added, or removed.
- [ ] **Database Schema Changes**: Indicate any database schema changes and their impact. Confirm that migration scripts were created.
- [ ] **Interface Version Changes**: Indicate any changes to interface versions.
- [ ] **Interface Dependencies**: Document added or removed dependencies.
- [x] **Permissions**: Document any changes to permissions.
- [x] **Logging**: Confirm that logging is appropriately handled.
- [x] **Unit Testing**: Confirm that changed classes were covered by unit tests.
- [x] **Integration Testing**: Confirm that changed logic was covered by integration tests.
- [x] **Manual Testing**: Confirm that changes were tested on local or dev environment.
- [x] **NEWS**: Confirm that the NEWS file is updated with relevant information about the changes made in this pull request.

### Related Issues
Implement Reindexing of Locations ([MSEARCH-702](https://folio-org.atlassian.net/browse/MSEARCH-702))

### CODE REVIEW

File: src/test/java/org/folio/search/service/IndexServiceTest.java [hunk 1]
```diff
@@ -265,6 +270,7 @@ void reindexInventory_positive_recreateIndexIsTrue() {
     verify(indexRepository).dropIndex(indexName);
     verify(consortiumInstanceService).deleteAll();
     verifyNoInteractions(locationService);
+    verifyNoInteractions(campusService);
```
Comment (BKadirkhodjaev) [inline src/test/java/org/folio/search/service/IndexServiceTest.java:273]: Probably not required but location-units are tied to inventory reindexing endpoint & service

File: descriptors/ModuleDescriptor-template.json [hunk 2]
```diff
@@ -57,7 +57,8 @@
             "inventory-storage.instance.reindex.post",
             "authority-storage.authority.reindex.post",
             "user-tenants.collection.get",
-            "inventory-storage.locations.collection.get"
+            "inventory-storage.locations.collection.get",
+            "inventory-storage.location-units.campus.collection.get"
```
Comment (BKadirkhodjaev) [inline descriptors/ModuleDescriptor-template.json:61]: Mirroring merged locations PR: https://github.com/folio-org/mod-search/pull/561/files

File: src/main/java/org/folio/search/service/IndexService.java [hunk 3]
```diff
@@ -154,11 +157,11 @@ && notConsortiumMemberTenant(tenantId)) {
       });
     }
 
-    if (LOCATION_RESOURCE.equals(resource)) {
-      return reindexInventoryLocations(tenantId);
-    } else {
-      return reindexInventoryAsync(resource);
-    }
+    return switch (resource) {
```
Comment (BKadirkhodjaev) [inline src/main/java/org/folio/search/service/IndexService.java:160]: Replaced to switch to accommodate other location-units sync reindexing

**Review by SerhiiNosko — APPROVED** [review]

**Review by azizbekxm — APPROVED** [review]

File: src/main/java/org/folio/search/service/locationunit/CampusService.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import org.folio.search.configuration.properties.ReindexConfigurationProperties;
+import org.folio.search.domain.dto.FolioIndexOperationResponse;
+import org.folio.search.domain.dto.ResourceEvent;
+import org.folio.search.service.ResourceService;
+import org.springframework.stereotype.Service;
+
+@Log4j2
+@Service
+@RequiredArgsConstructor
+public class CampusService {
```
Comment (psmagin) [inline src/main/java/org/folio/search/service/locationunit/CampusService.java:21]: It makes sense to combine it with LocationService. As well as CampusesClient combine with LocationClient. The logic of reindexing these entities is the same, the difference is only in HTTP requests.

File: src/main/java/org/folio/search/service/IndexService.java [hunk 5]
```diff
@@ -173,9 +176,17 @@ public ReindexJob reindexInventoryAsync(String resource) {
   }
 
   /**
-   * Runs synchronous locations reindex in mod-search.
+   * Runs synchronous locations and location-units reindex in mod-search.
    */
-  public ReindexJob reindexInventoryLocations(String tenantId) {
+  public ReindexJob reindexInventoryLocations(String tenantId, List<String> resources) {
+    resources.forEach(resourceName -> {
```
[the remark this answers is not in the document]
Comment (BKadirkhodjaev) [reply src/main/java/org/folio/search/service/IndexService.java:182]: We have 3 location-units: institutions, campuses and libraries

[the remark this answers is not in the document]
Comment (BKadirkhodjaev) [reply src/main/java/org/folio/search/service/IndexService.java]: Okay I just understood what you told me

**Review by psmagin — APPROVED** [review]

**Pull request merged**